services/storage/src/simcore_service_storage/s3_utils.py

```diff
@@ -1,14 +1,42 @@
 import functools
 import logging
+from typing import Final
 
 from botocore import exceptions as botocore_exc
-from pydantic import ByteSize
+from pydantic import ByteSize, parse_obj_as
 
 from .exceptions import S3AccessError, S3BucketInvalidError, S3KeyNotFoundError
 
+# this is artifically defined, if possible we keep a maximum number of requests for parallel
+# uploading. If that is not possible then we create as many upload part as the max part size allows
+_MULTIPART_UPLOADS_TARGET_MAX_PART_SIZE: Final[list[ByteSize]] = [
+    parse_obj_as(ByteSize, x)
+    for x in [
```

Comment:
MINOR: single char variables is ok, but pleeeeassse use other than ``x``! Perhaps the first initial as ``s`` from size?